### ISSUE TITLE
Summarizing detection data by Site/Location

### ISSUE THREAD

**Issue by j-sayers** [MotusDev/Motus-TO-DO#698]
This is a very big feature request that I've been thinking about for a while. Our recent conversation about [prospective sites](https://github.com/MotusDev/Motus-TO-DO/issues/696), and the definition and use of "sites" got me thinking again.

This relates to how detection data is displayed on the Motus, especially with regards to landowners, education partners, a member of the general public who likes to see what birds are detected on the nearby tower, though it applies to anyone who is viewing detection data on the website.

Basically the problem comes down to this: detection data on the website is summarized either by receiver (receivers can and do move) or by deployment (it is not uncommon for the same site to have multiple deployments over time). 

Currently the detections table for a deployment is the simplest and most straightforward way to share this information with pretty much anyone, so they can see at a glance what's been going by. It's also where I go to get a snapshot of what's been going on. 

But this is only straightforward if there has been only one deployment at that site. That's actually somewhat uncommon, especially for sites that have been in one location for a long time. Anytime the properties of the station change (different receiver, change of antenna bearings, change of antenna numbers) a new deployment should be created in order to capture the change.

**The ability to summarize by site/location over time would go a very long way towards making detection data more accessible on Motus.org**

There's a problem, though. Currently sites are not defined in that way. They are not necessarily discreet or attached to just one receiver at a time, nor are they necessarily georeferrenced.

I'm not sure how many people are using the concept of site in this more nebulous way, but the point is that without constraining its use, we can't use "sites" to refer to a general location that might have had multiple deployments over time.

I really think this is a pretty important feature (hopefully you all agree!) but the question is how to implement it.

A couple ideas come to mind, though there are probably others:

- Restrict the definition and use of "sites" to fit this concept (assuming the more nebulous use isn't in very wide usage and this change wouldn't have much implications)?

- Summarize detection data on the web by a buffered location, in addition to deployment and receiver?

Curious what your thoughts are...

**Comment by adamdsmith** [comment]
A good feature to have, speaking as one who often has to summarize data for a given location over multiple, and occasionally many, receiver deployments.

Presumably every  deployment is tied to a site and landowner, no? Can those not serve as two obvious levels of aggregation for detection summaries?

**Comment by jsayers-bsc** [comment]
> Presumably every deployment is tied to a site and landowner, no? Can those not serve as two obvious levels of aggregation for detection summaries?

In most cases, perhaps. Looks like that's how you are using it, and I noticed in one of the code snippets you shared previously it appeared you're using that site field to group your deployments in R, which is a great idea.

But it's not required that a site be defined this way. A "site" could be as large as a national park and contain multiple receivers, depending on what was desired. I'm not sure how many people do use sites in this way, though... If there are sufficient people using "sites" in this broader sense, then changing and restricting the definition would be problematic.

**Comment by adamdsmith** [comment]
Ah, got it. It is also complicates thing when a site is shared across projects, which we'll have soon when I transfer a receiver to a new project.

**Comment by StuMackenzie** [comment]
Is this something we really want to allow?

**Comment by adamdsmith** [comment]
The transfer? It's happened before that receivers have switched projects, no? I'll be ending the deployment on one project, transferring the receiver/station to its new owner/maintainer, and initiating a new deployment of the same receiver/site under the new project.

**Comment by leberrigan** [comment]
As I've been working on the new dashboard it has become clearer to me that organising things by receiver sites (or stations) is the way to go. 

Collaborators don't care about deployments, they're just a nuisance, but still important to keep track of because metadata. When introducing people to Motus there's always confusion around the purpose of deployments that takes time for people to understand and I think it's possible to avoid collaborators dealing with them at all. That would help ease some of the learning curve with Motus.

Instead, we can keep track of deployments in the background, whereas collaborators need only worry about which device is where. When they make changes to the station like change antenna orientations or height that gets reflected in the database as a new deployment. If a receiver is dismantled collaborators could chose to 'deactivate' the station so it gets reflected on the receiver map. 

Unfortunately, this might require making a _fourth_ table for receiver stations. This would fit in with the rest of the tables we have as follows:
 - **Receiver serial no.** : Unique for every physical computer 
 - **Receiver device ID**: Unique for every computer and project combo
 - **Station ID**: Unique for every physical point location (only one receiver can be at each station).
 - **Site ID**: Identifier for a cluster of receiver stations
    - We may need to think harder about how we define a site.

One good reason to do this is that right now it's not currently set up to easily keep track of changes to deployments that don't result in a termination. Just a few examples:
 - Antennas change orientation due to wind.
 - Telescopic sections of the mast collapse, significantly lowering the antenna height.
 - The station blows over or the mast breaks.

This would be something that is fixed immediately upon re-visiting the station so that when you leave it's in the same condition it was in when it was first deployed. But that means there is no record of the station condition before you fixed those issues and who knows how long it was in that altered state.

As for 'sites', we could make an option for collaborators to allow people from other projects see that they're available to encourage cross-project collaboration. This is especially helpful for areas where the network is expanding since it would allow people to keep track of new potential sites for deploying receivers. This obviously raises privacy concerns so we may want to require an upload of a signed consent form for each landowner that has their properties online.

**Comment by jsayers-bsc** [comment]
"Station" makes a lot of sense for how to define this. It conserves the colloquial word we use to refer to any receiver or deployment across time as long as it's at the same general location (Motus Station).

And it allows "Site" to be maintained in more or less its current context — no geographic information, no constraints on how many "stations" may be at a particular "site", where landowner information is stored...

Yes, it would require a new table, and some other changes on the back-end, but it would add a **lot** of clarity and ease of use to pretty much everyone, especially when it comes to dashboards and quick access of detections for outreach, education, and other partners.

**Comment by jsayers-bsc** [comment]
Another benefit for management is that it would allow deployment properties to be copied over from the previous deployment at the same station, regardless of whether the receiver has been changed.

**Comment by Hussell** [comment]
Notes from our last call. Discussion of the concepts and names for the concepts separated.

#### Concepts:
 * A named location with a land owner. E.g. provincial parks, private properties. Usually polygons. Perhaps with a point location and radius so we don't have to store polygons.
 * A named location with a series of temporally non-overlapping deployments. Usually a point location, perhaps with a small radius to encompass minor changes in deployment location.

 Owned land can have zero, one, or many parallel deployment-locations.

 A deployment-location has zero or one owned-land location.
  * Zero for proposed or planned locations, where the land-owner might not be known.

 A deployment-location can have zero, one, or many serial deployments.
  * Zero for proposed or planned locations.
  * One-or-more for real, stationary receivers.

A deployment has zero or one deployment-location.
  * Zero for mobile receivers (aircraft, cars, ships, hand-held).

A deployment has exactly one project, but deployment locations and owned land can be used by multiple projects (serially and in parallel, respectively).

A deployment won't need a name any more. (But we probably won't remove it from the database for a long time.)

Does a deployment need its own latitude and longitude, or can we rely on the deployment-location + GPS for mapping?

#### Names:

What we (I?) want, descending priority:
* intuitiveness for researchers (so, avoid jargon, avoid subtly redefining existing words, strive for clarity, specificity respectively)
* brevity
* ease of translation into French, Spanish

Proposed names:
 * site, station
 * site group, site
 * property, station
 * brainstorm more ideas

**Comment by leberrigan** [comment]
Thanks for laying this out, Jeremy. This looks good to me. I have one point to add:
- It would be useful if a named location with a landowner could have the option to be: shared between all projects (any logged in user can see it); shared between select projects (only logged in users of certain projects can see it); or private (only visible to logged in users of a single project). 

**Names**
There are two different identifiers we're talking about here: 
- Named location of an area (polygon) with a landowner: 
  - This is equivalent to 'site' right now and I don't see much of a reason to change it.
- Named location of a point with temporally non-overlapping deployments. 
  - This does not currently exist, but we have proposed to organise all data based on the named point instead of deployment or receiver (computer). 
  - Whatever term we chose to use here will likely become the most-commonly used term we used to refer to a Motus data source by researchers and the public. 
  - Currently, I would say the most commonly-used term is 'tower' and I think we can all agree that is not a good word to use considering the many variations that don't involve a tower structure, plus the fact a tower is only used to mount antennas.
  - I'm holding my vote for 'station' since I think this is the _second_ most commonly-used term by the public and collaborators. To me, this is intuitive because it's like a radio station and I think this intuition will exist across languages: **français**= station; **español**= estación or emisora.

**Comment by Hussell** [comment]
> Named location of an area (polygon) with a landowner:
> This is equivalent to 'site' right now and I don't see much of a reason to change it.

'Site' is usually a term for a place where something (usually one thing) is located, and we're proposing to use it here to mean something else. If we use 'site' that will become a new piece of Motus jargon, with a specific meaning for us which is not obvious to users.

Perhaps 'property', in the sense of real-estate or more generally 'something owned'? Are there cases where the boundaries of an area don't conform to property lines? 'Area' might better convey "can contain multiple stations", if 'property' isn't sensible.

> I'm holding my vote for 'station' since I think this is the second most commonly-used term by the public and collaborators. To me, this is intuitive because it's like a radio station

Receiver station, antenna station, Motus station. Sounds/feels OK to me. (Although a radio station was usually a broadcasting transmitter rather than a receiver. Perhaps the meaning has drifted recently, e.g. a GPS or network 'base station' is usually a receiver.)

**Comment by jsayers-bsc** [comment]
Glad to see this is moving along!

A few additional thoughts...

- Named location of an area (polygon) with a landowner: 
  - These currently don't have any location information associated with them. Maybe they don't need to?
  - We may end up with confusion between "site" and "station". Maybe "Property" would be clearing for this
  - Ability to be associated with more than one contact. Currently there is only one user associated with a site -- Landowner -- which isn't ideal in many cases (#628)

- Named location of a point with temporally non-overlapping deployments. 
  - I definitely agree that "station" is the most suitable term for this. It's what most people call it anyway so it's going to make a lot on intuitive sense.
  - the Station level also makes the most sense for management  also makes most sense for the primary point where changes are made
    - Change receivers or modify antenna configurations (creates a new deployment record in the back-end, which users probably don't need to see)
    - Save comments and notes related to site visits that will persist across deployments 
    - A table on the Stations page summarizing changes in Station properties over time (e.g. deployment records)
- Stations would be defined by the user and associated with one deployment record at a time by way of a stations_deployments or similar table, so lat/lon for a station would only be for the purpose or visual display and not linking stations to deployments (so no need for a radius in a station definition). 
    - A mobile motus station wouldn't have a location at all, but otherwise we'd probably want to enforce defining a station location for mapping purposes, even though the actual location will be in the detection data
- Do we allow a station to be moved? In most cases moving a station would be the same as terminating one and starting another, though there are cases where a station is moved around locally but still considered the same station. Maybe we just set a limit on how far it can move and still be the same station?

**Comment by leberrigan** [comment]
> Site' is usually a term for a place where something (usually one thing) is located, and we're proposing to use it here to mean something else. If we use 'site' that will become a new piece of Motus jargon, with a specific meaning for us which is not obvious to users.

To me a site can be an area, such a study site, but I do agree that it's generally used to define a single point. We should just be careful about changing names around unless we're certain it needs to happen.

If we are to change names I think 'property' would be fitting for most situations, but we might want to find a more generic term like 'area' or 'locality'. 

> Do we allow a station to be moved? In most cases moving a station would be the same as terminating one and starting another, though there are cases where a station is moved around locally but still considered the same station. Maybe we just set a limit on how far it can move and still be the same station?

I'm not sure about this one, but it raises a good point. I think going back to Jeremy's original definition, I think a 'station' or whatever should be a point with a given radius, allowing for some variation in where exactly it's placed. I'm trying to imagine what kind of implications we'd get from having a radius for a station and I can think of a couple:
  - The point of a station is to match a series of deployments to a single location so I think it doesn't make sense to allow them to be moved.
  - Matching receiver GPS to 'station' location would help us validate lat/lng. For instance, we could create an alert for when GPS falls far outside of expected location.
  - Sometimes receivers are deployed in a dense grid for fine spatial resolution (think CTT Sensor Nodes) so we should limit the station radius to something much smaller than the highest density of stations that we'd expect (max 100 m?).

**Comment by denislepage** [comment]
I missed the fun discussion today, so my opinion may come too late.

I would agree that a deployment would happen at a station, which would be a new concept that we don’t have now and that represents a single point.

What we have called site so far is ambiguous, and I would advocate we find a new name. I am recommending “area”. People have been using the “site” in place of “station” at times, and as an “area” at other times. Area and station are respectively more reflective of a polygon and a point concept, I think.

We could potentially allow people to draw their area on a map, or we can draw ourselves from the cumulative stations data. Allowing people to draw their own would serve the need for “potential” sites (oops, areas). Showing polygons for areas may help clarify the distinction in any case.

Now, this brings many questions:


1.       Do we require all stations to be attached to an area, like we do now for deployments vs. sites? I vote yes, but we will need to build an interface that easily allow handling the common cases where an area only has 1 station.



2.       Do we provide recommendations about how large an area should be?  I guess this ties to the purpose of the area concept, so we’ll need a clearer idea of how we intend to use this ourselves and will need to explain it somewhere.


3.       Do we require all deployments to be attached to a station? (I vote yes, perhaps including mobile, see next)


4.       How about mobile deployments? We still want starting coordinates, and I think I would prefer to move them to station and have mobile stations, rather than mobile deployments, so we don’t have to keep coordinates in both tables. I guess this also implies that the type of mount (e.g. tower, building) would be a property of stations, not deployments? Not sure if that makes sense.


5.       How do we characterize the areas spatially? I lean towards option 1.



o   Option 1: we define areas with a polygon, and we calculate a centroid spatially (no need to ask users, but we can cache it whenever the polygon changes). When showing the areas on the receiver map at high resolution, we switch to showing station location and areas as polygons. Track data would only show station locations. If we need exporting area location as tabular data, we use the centroid, but I am not convinced this is needed.



o   Option 2: we ask people only for the centroid of their areas.


6.       Do we allow people to redraw an area or change the centroid? (I vote yes)


7.       Do we allow people to change a station location? (I vote yes, but this would apply to all previous deployments)


8.       Should tag deployments follow the same idea and be attached to stations? (I vote no).


9.       Do we allow concurrent deployments of distinct receivers at the same station? (I vote no).


10.   Do we allow concurrent deployments of distinct receivers in the same area? (clearly yes)



11.   Do we allow stations to be moved to different sites (I vote yes).



12.   Can we have more than 1 station at the same coordinates, but with different properties (e.g. tower vs. hand-held). We probably don’t need to worry about this, but we may want to have a merge function for station.


13.   Concurrent deployments of the same receivers should never be allowed among stations, areas or projects, as we have now.


14.   Are areas only available within projects, or can they be shared across projects (e.g. based on user permissions, which makes this more complicated)? I vote no. Unlike tags, the main purpose of assigning deployments and areas/stations to projects is for managing receivers, and we would encourage grouping receivers management based on responsibility for maintenance, and I can’t think of a good enough reason to allow the same site to be used for multiple projects.



15.   Presumably, all metadata records would be attached to the location of the station, not the area.

Here’s my view on properties stored at the various concept levels:

Sensors Hash Keys: list of unique serial numbers across all projects. No changes needed.
Sensors: list of receivers within each project (receiverID, projectID). No changes needed.
Sensor Deployments: start and end datetimes, antenna config, node details and station ID (or mobile). Changes needed would be renaming siteID to stationID, and the fact that we no longer want coordinates (moved to stations table).
Stations: coordinates, name(?), area ID (new table). Does not need a projectID if always assigned to an area, itself contained in a project.
Areas: centroid, polygon, name, landowner ID, project ID (more or less repurposing the sensor_sites table, which I would rename project_areas and basically would just need an extra field for the polygon)
Landowners: contact, etc.

At this point, I am assuming that any changes in sandbox (Java or db) relate to beta10, though it’ll likely be a while before we’re ready to make that transition, and we’ll need a way to bridge the differences when we’re ready to deploy beta10 to beta (e.g. keeping both the siteID and stationID fields in deployments for now, and keeping the sensor_sites together with projects_areas and turning off some db requirements like having a stationID in deployments).

I’d advocate trying to deploy a first beta10 with relatively limited new features, so we don’t have to deal with too many of these types of issues at once, which may mean not waiting for the entire dashboards to be ready before we release the above changes.

Something to think about.

**Comment by Hussell** [comment]
> Do we require all stations to be attached to an area, like we do now for deployments vs. sites?

I'm inclined to allow some slack here for any project that doesn't care to track contact information for an area. E.g. an exploratory university project which sets up one tower on their own building, or a mobile deployment on a car traveling on public highways.

> Do we require all deployments to be attached to a station?

> How about mobile deployments? We still want starting coordinates, and I think I would prefer to move them to station and have mobile stations, rather than mobile deployments, so we don’t have to keep coordinates in both tables. I guess this also implies that the type of mount (e.g. tower, building) would be a property of stations, not deployments? Not sure if that makes sense.

Discussion on the call went a bit in the other direction. A station is a named location, but we've had deployments moved a small amount several times, e.g. different corners of someone's back-yard, or a tower that fell over due to wind and was moved ~20m, but was treated as part of the same sequence of deployments. So we were thinking (or I was, at least) that deployments should still have their own latitude and longitude, and we'd set up some data entry warnings and data issue notifications if they were too far (>100m?) from the station latitude and longitude. Given that, it seems more sensible for mobile deployments to not have a station. On the other hand, doing it that way would leave mobile deployments without a name...

Maybe give stations a name, point location, and radius (default of 0m or some very small radius), remove the name and point location from deployments, and make it mandatory for deployments to have a station. Then mobile deployments can have a name, and be given a central point-location and a large-enough radius to encompass the entire path they took. That seems more useful than starting location. (Currently the maps do not display mobile deployments at all, because many of the starting locations are confusing when mixed in with stationary stations.) The rare stations which get moved a little bit can adjust the station's point location and increase the station's radius. If a station has an imprecise GPS, the radius can be raised to get rid of data issue notifications about GPS hits too far from the station. It would be easy (for me, I think) to write a little bit of code which automatically generates the minimal circle which encompasses all the GPS hits during a deployment, to make things a little easier for users, especially those making mobile deployments. (Alternately, the median GPS location, and a radius which encompasses all but outlier locations, with automatic detection of outliers.) I'm not anticipating displaying the station radius on any public maps, only the point location at the center.

One of the things Josh mentioned is that there have been cases where two different receivers were deployed at the same location at the same time. I'm not sure if those had different fixture types too, but it seems like they could be different, theoretically. E.g. one way-up on a radio tower and another closer to the ground on a pop tower or a house.

> How do we characterize the areas spatially?

(As a polygon or as a circle?) Polygons can potentially be very complicated, but so long as all we're doing is displaying them, that's mostly storage space not CPU time. Developer time for polygons might be 2 to 10 times as much? (Google maps has a polygon drawing library, and we'd probably need KML, geoJSON, or GIS import for polygons.) This seems like something where it would be relatively easy to start with circles and switch to polygons later, or vice versa if polygons turn out to be too onerous. I guess I have no preference, since I'm not sure how valuable polygons for this will be to users.

> Do we allow people to redraw an area or change the centroid?

> Do we allow people to change a station location?

Another yes for both. No reason not to.

> Should tag deployments follow the same idea and be attached to stations?

Not just no, but clearly no. The majority of tags are not deployed at a receiver station. Unless this means we should consider setting up named banding locations? A lot of tags do share the same deployment location with other tags. I'm not sure why this would be desired, though, other than to save a little space in the database.

> Do we allow concurrent deployments of distinct receivers at the same station?

Josh mentioned that this is something that has happened, though rarely. We do, and should continue to, enforce the requirement that receiver hardware (with a unique serial number) cannot have overlapping deployments, but I think having overlapping deployments of different receivers at the same station should generate a warning or confirmation request, at most.

> Do we allow stations to be moved to different sites?

Seems like something which wouldn't be necessary very often except to correct mistakes, but no reason not to allow it. (Do we allow overlapping areas? I guess we won't care much about that so long as a station is linked to exactly one area.) We should probably make sure the point location of a station is inside, or at least nearby, the area.

> Can we have more than 1 station at the same coordinates, but with different properties (e.g. tower vs. hand-held)?

If we allow overlapping deployments at the same station for different receiver serial-numbers, then we could enforce non-overlapping stations. If we don't, then multiple stations at the same location will probably be how people represent two different receivers deployed at the same time and location, which, once again, has happened and will likely continue to happen. Disallowing both would just be mean. My preference is to make station locations unique, and let deployments of different receivers at the same station overlap (but not of the same receiver, obviously).

We should also consider enforcing station name uniqueness within the same project, and perhaps within the same area or globally if we allow areas and stations to be used by multiple projects.

> Are areas only available within projects, or can they be shared across projects?

I vote yes, and for stations too. There are a number of groups which have set up serial projects at the same location, i.e. a lab run by a senior researcher, with a study area where receivers are set up, and different projects run by different graduate students in different years. (For example the projects named "Helgoland Spring 2017" through "Helgoland Autumn 2020".) Another example is control of a study area shifting to another lab or researcher focusing on a different study animal, since projects are often named things like "NCC Shorebird Monitoring Project" or "Taylor - Swallows".

Projects are a pretty flexible way to group people, receivers, and tags together over time. Some people are naturally going to want to divide things up into lots of small projects, while others will want one big project. (Based on observing existing behaviour on Motus and iNaturalist.) Some people will divide them by year or season, others by researcher or institution, others by location, others by study animal. I don't think there's any particular reason to force people to do things one way or the other, other than our own natural inclinations. Avoiding making any of these organizational principles inconvenient seems like one good goal to include in this enhancement. So, anything physical can be associated with multiple projects at different times, and depending on the thing, at the same time. This includes collaborators, receivers, tags, and also locations.

**Comment by Hussell** [comment]
Well, I feel ambitious, so here's something ambitious. Don't take the names too seriously, the important things are the structure and constraints.

Table For:|Columns For:
-|-
Receiver Deployments|project ID, receiver ID, station ID (optional), name (optional), notes, start and end date, fixture type, etc.
Receivers|serial number, manufacturer, model, hash key (optional)
Stations|name, notes, latitude & longitude, radius, optional area ID
Areas|name, notes, latitude & longitude (centroid) or multipolygon or both
Area-Contacts|area ID, contact ID (user ID), notes

Under this scheme/schema:
 - Stations, areas, receivers won't need be duplicated to allow them to be used by another project.
   - sensors_hash_keys and sensors could be merged. (But don't have to be.)
   - Stations, areas, and receivers will initially be created without being linked to any project. (Project 0 will be obsolete and new locations can be planned without creating a project.)
 - Stations won't need to be duplicated when multiple receivers are deployed at one place at the same time.
 - Stations won't need to be duplicated when there are deployments with different fixture types at the same location.
 - Areas won't need to be created for stations which don't need one.
 - Stations won't need to be created for deployments which don't need one, e.g. most mobile deployments.
   - Deployments can still be given individual names if they don't have a station, or if otherwise desired.
 - Free-form notes everywhere. These get used for astonishing things, but support for them is missing in many places, e.g. #31.
 - Areas will be able to have as many and as few contacts as necessary.
   - When adding or removing contacts, won't have to figure out which columns are empty.
   - When listing or searching contacts, won't have to query empty columns.
   - Easier to ensure uniqueness of contacts for an area (key constraint instead of comparing columns).
   - No arbitrary upper limit.
 - Contacts could be put in their own table (currently they're mixed in with user accounts), but don't have to be. This could open the way to having, say, any number of phone numbers, emails, and addresses for each contact, which is currently impossible unless we either change every user table at BSC at once or stop reusing BSC user code in Motus.

The biggest difference from the way things are and from other proposals: deployments will have a project ID, and receivers, stations, and areas won't. In practice project IDs will still be stored in other tables, but those columns will be deprecated. Stations and areas will be optional organizational tools, but almost everything will be summarized by station so there will be a lot of incentive to use them.

Issues:
 - Some common queries will be slightly longer (but equally efficient), others will be shorter. No obvious net benefit or loss, I think, even after taking into account joins.
 - Some queries will be possible, and necessary, that were formerly difficult or impossible, e.g. searching all stations when deciding whether to link an existing one to a deployment or create a new one.
 - It might be helpful to generate default deployment names by appending a number to a station name, e.g. station name is "Old Cut", deployment names are "Old Cut 1", "Old Cut 2", etc., or by appending dates "Old Cut Jan. to May 2020", "Old Cut May to Aug. 2020", etc. Or it might just be annoying. I'm not sure.
 - Notes should be visible to project collaborators, but only shared with people outside the project if the "Share All Data" option is chosen.
 - Contact info shouldn't be visible outside a project. A difficulty here is that areas can be shared between projects, so a malicious user could create a project, create a station in any existing area, then view the contact info for that area.
   - Rejected solution: go back to not sharing areas between projects. Rejected due to making it necessary to duplicate areas for multiple projects, which also disincentivizes creating new projects in the same area.
   - Possible solution: a project ID column in the Area-Contact table. Maximum flexibility.
   - Possible solution: an area/project table. An area's contacts can be seen by some but not all projects. Not possible for different projects to have different contacts for an area.
   - Preferred solution: a contact/project table. Individual contacts can be seen by some but not all projects.  Not possible for contacts to be linked to different areas for different projects. (We have this table already: projects_landowners. But there's no method to search for contacts in other projects a user has permissions with, so projects in the same area have to duplicate contacts.)

P.S.:

I'd also like to have a project ID column in the tag deployments table.
 - Tags won't need to be duplicated to allow them to be used by another project.
 - Tags can initially be registered without being linked to any project. (Project 0 will be obsolete, and Lotek won't have to know (or guess) which project a tag belongs to.)

**Comment by Hussell** [comment]
A few examples of mobile deployments (make sure you log in):
https://motus.org/data/receiverMap?receiverDeploymentId=947
https://motus.org/data/receiverMap?receiverDeploymentId=3699
https://motus.org/data/receiverMap?receiverDeploymentId=3771

**Comment by cjardine-bsc** [comment]
Heya Kids, 

The sandbox database should now be set up to handle the new sites-stations-deployments structure. 

You'll see there are two new tables: sensor_sites_stations and sensor_sites_landowners. 

I've moved landowner information from the sensor_deployments and sensor_sites tables over to sensor_sites_landowners (many to many relationship) and move station-level information from sensor_deployments over to sensor_sites_stations (one to many relationship to sensor_deployments). 

I've gone through the database and made changes anywhere the sensor_deployments table was referenced and fixed any errors moving the fields had caused. 

One note, stations and sites will be project specific, not shared across projects. 

Next steps:

1. Create a new station metadata management page in sandbox. (mock up/concept by @StuMackenzie, @jsayers-bsc, @leberrigan. Implementation by @pmorrill and @Hussell 
2. Update landowner management page to allow selecting multiple landowners per site, and removing landowners from the deployment records. (mock up/concept by @StuMackenzie, @jsayers-bsc, @leberrigan. Implementation by @pmorrill and @Hussell 
3. Go through the web and java code for sandbox and clean up anything that was broken when I rearranged the tables. @Hussell 
4. I'll want to expand the data validation proc to include some checks with respect to the new table. @cjardine-bsc 

I think that's all we'll want to tackle before we cut this off as a beta 10.
